packages/ec-core/src/operator/mutator/mod.rs

```diff
@@ -2,14 +2,197 @@ use rand::rngs::ThreadRng;
 
 use super::{Composable, Operator};
 
+/// Mutate a genome of type `G` generating a new
+/// genome of the same type (`G`).
+///
+/// Implementations of this trait are typically representation dependent,
+/// so see crates like [`ec_linear`](../../../ec_linear/mutator/index.html) for
+/// examples of mutators on linear genomes.
+///
+/// # See also
+///
+/// See [`Mutate`] for a wrapper that converts a `Mutator` into an [`Operator`],
+/// allowing mutators to be used in chains of operators.
+///
+/// # Examples
+///
+/// In this example, we define a `Mutator` that flips one random bit in a
+/// `Vec<bool>`. We then use this `Mutator` to mutate a genome, and check that
+/// exactly one bit has changed.
+///
+/// ```
+/// # use rand::thread_rng;
+/// # use rand::rngs::ThreadRng;
+/// # use rand::Rng;
+/// # use ec_core::operator::mutator::Mutator;
+/// #
+/// struct FlipOne;
+///
+/// impl Mutator<Vec<bool>> for FlipOne {
```

Comment:
Do we want `Vec<bool>` or `[bool;4]` in the example here? Since I think we don't really need a vec and in that case an array would be more ideomatic

Reply:
We started using arrays on some of the tests late in the game, and I meant to come back and convert these and obviously forgot. Thanks for catching that.

If `FlipOne` was a "real" mutator, would we instead take `&[bool]` instead of either a vector or an array? Should we write `FlipOne` that way here? It complicates things a fair bit (some lifetime weirdness), but it can be done.

Reply:
We could probably take either &mut [T] or some sort of impl DerefMut<[T]> which would be more generic and allow us to pass vectors as well and not just references

Reply:
I'd forgotten about `DerefMut<[T]>`, but that would definitely be better. It's late, so I'm going to commit what I have, but I'll try to see if I can convert over to `DerefMut<[T]>` tomorrow.

Reply:
In the interest of moving this forward, I think it makes sense to push the whole `Deref` thing off to another day and get this PR merged in.